src/main/java/com/spotify/docker/client/DockerConfigReader.java

```diff
@@ -56,63 +44,174 @@
   }
 
   /**
    * Returns the RegistryAuth for the config file for the given registry server name.
    *
    * @throws IllegalArgumentException if the config file does not contain registry auth info for the
    *                                  registry
+   * @deprecated In favor of {@link #authForRegistry(Path, String)}
    */
+  @Deprecated
   public RegistryAuth fromConfig(final Path configPath, final String serverAddress)
       throws IOException {
-    return parseDockerConfig(configPath, serverAddress);
+    return authForRegistry(configPath, serverAddress);
   }
 
   /**
-   * @deprecated do not use - only exists for backwards compatibility. Use {@link #fromConfig(Path)}
-   *     instead.
+   * Return a single RegistryAuth from the default config file.
+   * If there is only one, it'll be that one.
+   *
+   * @return Some registry auth value.
    */
-  @Deprecated
-  public RegistryAuth fromFirstConfig(Path configPath) throws IOException {
-    return parseDockerConfig(configPath, null);
+  public RegistryAuth anyRegistryAuth() throws IOException {
+    return anyRegistryAuth(defaultConfigPath());
   }
 
-  private RegistryAuth parseDockerConfig(final Path configPath, final String serverAddress)
-      throws IOException {
+  /**
+   * Return a single RegistryAuth from the config file.
+   * If there are multiple RegistryAuth entries, which entry is returned from this method
+   * depends on hashing and should not be considered reliable.
+   * If there is only one entry, however, that will be the one returned. This is the
+   * primary use of this method, as a useful way to extract a RegistryAuth during testing.
+   * In that environment, the contents of the config file are known and controlled. In a
+   * production environment, the contents of the config file are less predictable.
+   *
+   * @param configPath Path to the docker config file.
+   * @return Some registry auth value.
+   */
+  @VisibleForTesting
+  RegistryAuth anyRegistryAuth(final Path configPath) throws IOException {
+    final ImmutableCollection<RegistryAuth> registryAuths =
+        authForAllRegistries(configPath).configs().values();
+    return registryAuths.isEmpty()
+        ? RegistryAuth.builder().build()
+        : registryAuths.iterator().next();
+  }
+
+  /**
+   * Parse the contents of the config file and generate all possible
+   * {@link RegistryAuth}s, which are bundled into a {@link RegistryConfigs} instance.
+   * @param configPath Path to config file.
+   * @return All registry auths that can be generated from the config file
+   * @throws IOException If the file cannot be read, or its JSON cannot be parsed
+   */
+  public RegistryConfigs authForAllRegistries(final Path configPath) throws IOException {
     checkNotNull(configPath);
 
-    final Map<String, RegistryAuth> configs = parseDockerConfig(configPath).configs();
+    final DockerConfig config = MAPPER.readValue(configPath.toFile(), DockerConfig.class);
+    if (config == null) {
```

Comment:
not very important, but can `ObjectMapper.readValue()` actually ever return `null`?

Reply:
I wrote this null check purely defensively. But, given your question, I dug into the source to figure this out. Yes, it is possible.

`readValue(File , Class<T>) → _readMapAndClose(JsonParser, JavaType)`. The latter does under some conditions set `Object result = null` and returns `result`.